hammers/scripts/clean_old_aggregates.py

```diff
@@ -63,21 +77,75 @@ def clear_aggregates(agg_list):
 
     return errors, report
 
+def orphan_find(allaggs):
+
+    # Find all hosts currently in aggregates
+    hosts_from_aggs = []
+    for agg in allaggs.values():
+        for host in agg['hosts']:
+            hosts_from_aggs.append(host)
+
+    # Make list of ironic hosts not in any aggregate
+    ironic_hosts = osrest.ironic.nodes(auth, details=False).keys()
+    orphans = []
+    for host in ironic_hosts:
+        if host not in hosts_from_aggs:
+            host_id = [h['id'] for h in osrest.blazar.hosts(auth).values() if h['uid'] == host][0]
+            orphans.append(host_id)
+    return(orphans)
+
+def has_active_allocation(orph):
+
+    matching_allocs = [alloc for alloc in host_allocs if alloc['resource_id'] == orph]
+    if not matching_allocs:
+        return False
+    res = matching_allocs[0]['reservations'][0]['id']
+    return(res)
+
+def del_expired_alloc(db, old_alloc):
+
+    ha_id = old_alloc['id']
+    l_id = old_alloc['lid']
+    hh = old_alloc['hypervisor_hostname']
+
+    msg=("Deleting host_allocation for host {} matching expired lease {}. ".format(hh,l_id) + "\n")
+    count = query.blazar_old_host_alloc_delete(db, ha_id)
+    return(msg)
 
 def main(argv=None):
+    
     slack = Slackbot(args.slack, script_name='clean-old-aggregates') if args.slack else None
 
     try:
         term_leases = [lease for lease in leases.values() if is_terminated(lease)]
         old_aggregates = [aggs for aggs in (aggregates_for_lease(lease) for lease in term_leases) if aggs != None]
         aggregate_list = list(itertools.chain(*old_aggregates))
-        errors, report = clear_aggregates(aggregate_list)
-
-        if report:
-            str_report = ''.join(report)
-
-            print(str_report)
-
+        errors, reports = clear_aggregates(aggregate_list)
+        orphan_list = orphan_find(aggregates)
+        reports = []
+        
+        if orphan_list:
```

Comment:
Looks like `orphan_list` is always going to be a list. Since the logic doesn't do anything special when it's empty, I'd just skip this check and proceed to `for orphan in orphan_list`, which effectively won't execute if the list has 0 items.